src/Image_Backend.php

```diff
@@ -84,87 +78,61 @@ public function setImageResource($resource);
      * will be calculated from the given data.
      */
-    public function writeToStore(AssetStore $assetStore, $filename, $hash = null, $variant = null, $config = []);
+    public function writeToStore(AssetStore $assetStore, string $filename, ?string $hash = null, ?string $variant = null, array $config = []): array;
 
     /**
      * Write the backend to a local path
      *
      * @param string $path
      * @return bool if the write was successful
      */
-    public function writeTo($path);
+    public function writeTo(string $path): bool;
 
     /**
      * Set the quality to a value between 0 and 100
-     *
-     * @param int $quality
      */
-    public function setQuality($quality);
+    public function setQuality(int $quality): static;
 
     /**
      * Resize an image, skewing it as necessary.
-     *
-     * @param int $width
-     * @param int $height
-     * @return static
      */
-    public function resize($width, $height);
+    public function resize(int $width, int $height): static;
 
     /**
-     * Resize the image by preserving aspect ratio. By default, it will keep the image inside the maxWidth
-     * and maxHeight. Passing useAsMinimum will make the smaller dimension equal to the maximum corresponding dimension
-     *
-     * @param int $width
-     * @param int $height
-     * @param bool $useAsMinimum If true, image will be sized outside of these dimensions.
-     * If false (default) image will be sized inside these dimensions.
-     * @return static
+     * Resize the image by preserving aspect ratio. By default, the image cannot be resized to be larger
+     * than its current size.
+     * Passing true to useAsMinimum will allow the image to be scaled up.
      */
-    public function resizeRatio($width, $height, $useAsMinimum = false);
+    public function resizeRatio(int $width, int $height, bool $useAsMinimum = false): static;
 
     /**
      * Resize an image by width. Preserves aspect ratio.
-     *
-     * @param int $width
-     * @return static
      */
-    public function resizeByWidth($width);
+    public function resizeByWidth(int $width): static;
 
     /**
      * Resize an image by height. Preserves aspect ratio.
-     *
-     * @param int $height
-     * @return static
      */
-    public function resizeByHeight($height);
+    public function resizeByHeight(int $height): static;
 
     /**
-     * Return a clone of this image resized, with space filled in with the given colour
-     *
-     * @param int $width
-     * @param int $height
-     * @param string $backgroundColor
-     * @param int $transparencyPercent
-     * @return static
+     * Return a clone of this image resized, with space filled in with the given colour.
      */
-    public function paddedResize($width, $height, $backgroundColor = "FFFFFF", $transparencyPercent = 0);
+    public function paddedResize(string $width, string $height, string $backgroundColour = 'FFFFFF', int $transparencyPercent = 0): static;
```

Comment:
Change `color` to `colour` - these are abstracted by the `ImageManipulation` trait anyway so most people proably won't be calling it directly, much less with named arguments.